src/utils/HalfEdgeNode.h

```diff
@@ -48,6 +48,21 @@ class HalfEdgeNode
         }
         return odd_path_count > 2;
     }
+    
+    bool isMarked() const
```

Comment:
Proposal: Merge `isMultiIntersection` and `isMarked` into a single function that checks whether the `isMarked` count of going around an edge is larger-or-equal than '2 or 1, respectively', which can be the argument. 

(We could also muck around with writing a sort of map-around-the-vertex function that takes a closure as argument, but that's probably overkill.) (Anyway, at least `isMultiIntersection` can also have an early-out and then return false just like `isMarked`.)